cornell-dti/office-hours
# New session information header

### Summary 


Implemented new session information header. TAs can now edit/update their virtual location at any time during office hours. Students can view the join zoom session button at all times but can only join once it is their turn. 

The zoom link displayed differs based on office hour modality. For virtual office hours, the TA sees "update virtual location" if they did not provide a zoom link. Once they provide a zoom link, they will have the ability to view/edit it. On the student's side, they will see a join button. If they click on it before filling out the form, before it is their turn, or after the office hour session has ended, an error message will pop up. For in person, "No Zoom Link Available" is displayed. For hybrid, "Use student provided Zoom link" is displayed (See note on the hybrid modality). For discussions/review sessions, the join button should be visible for both the TA and students, and it should take them to the review session link. 








### Test Plan 



- Sign in as a professor/TA, update your virtual location with a **valid** URL. 
- Sign in as a student. Try joining the zoom session before filling out the join the queue form (should get an alert). Join the queue. Try joining the zoom session before you are assigned to a TA (should get an alert). Once you are assigned to a TA, join the zoom session. This should take you to the TA's virtual location. Update the TA's virtual location, and then try joining the session as a student (it should take you to the new location). If the TA removes their location after assigning a student to them, when the student tries to join the zoom session they should get a message telling them to wait for the TA to update their location.
- Create and test different office hour modalities. 

### Notes 


I think there is a bug with the hybrid modality. When filling out the join the queue form, the student provides their location. However, once they join the queue the location is not visible to the TA nor the student. The student must click on the location icon and provide their location again in order for it to be visible to the TA (below the students name).

### Breaking Changes 

None
